src/Sarif.Converters/TSLintConverter.cs

```diff
@@ -0,0 +1,147 @@
+            {
+                Name = "TSLint"
+            };
+
+            output.WriteTool(tool);
+
+            var results = new List<Result>();
+            foreach(TSLintLogEntry entry in tsLintLog)
+            {
+                results.Add(CreateResult(entry));
+            }
+
+            var fileInfoFactory = new FileInfoFactory(MimeType.DetermineFromFileExtension, loggingOptions);
+            Dictionary<string, FileData> fileDictionary = fileInfoFactory.Create(results);
+            
+            if (fileDictionary?.Any() == true)
+            {
+                output.WriteFiles(fileDictionary);
+            }
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+        }
+
+        internal Result CreateResult(TSLintLogEntry tslEntry)
+        {
+            tslEntry = tslEntry ?? throw new ArgumentNullException(nameof(tslEntry));
+
+            Result result = new Result()
+            {
+                RuleId = tslEntry.RuleName,
+                Message = tslEntry.Failure
+            };
+
+            switch (tslEntry.RuleSeverity)
+            {
+                case "ERROR":
+                    result.Level = ResultLevel.Error;
+                    break;
+                case "WARN":
+                case "WARNING":
+                    result.Level = ResultLevel.Warning;
+                    break;
+                case "DEFAULT":
+                    result.Level = ResultLevel.Default;
+                    break;
+                default:
+                    result.Level = ResultLevel.NotApplicable;
+                    break;
+            }
+
+            result.Locations = new List<Location>();
+
+            Region region = new Region()
+            {
+                // The TSLint logs have line and column start at 0, Sarif has them starting at 1, so add 1 to each
+                StartLine = tslEntry.StartPosition.Line + 1,
+                StartColumn = tslEntry.StartPosition.Character + 1,
+                EndLine = tslEntry.EndPosition.Line + 1,
+                EndColumn = tslEntry.EndPosition.Character + 1,
+
+                Offset = tslEntry.StartPosition.Position
+            };
+
+            int length = tslEntry.EndPosition.Position - tslEntry.StartPosition.Position + 1;
+            region.Length = length > 0 ? length : 0;
+
+            Uri analysisTargetUri = new Uri(tslEntry.Name, UriKind.Relative);
+
+            PhysicalLocation analysisTarget = new PhysicalLocation(uri: analysisTargetUri, uriBaseId: null, region: region);
+            Location location = new Location();
+            location.AnalysisTarget = analysisTarget;
+
+            result.Locations.Add(location);
+
+            result.Fixes = new List<Fix>();
+
+            if (tslEntry.Fixes?.Any() == true)
+            {
+                IList<Replacement> replacements = new List<Replacement>();
+
+                foreach (TSLintLogFix fix in tslEntry.Fixes)
```

Comment:
To understand if this code is right, I need to understand the semantics of `Fixes`. Is the idea that, to correct the problem identified by the log entry, you must make _all the replacements specified by the Fixes_? In other words, does a TSLint "fix" have the same semantics as a SARIF "replacement"?

Reply:
Couldn't find a spec, so I inferred from the logs I had. Semantics seem to fit. 
- innerStart aligns with startPosition.position (which is a 0-initiated offset into the file). So it fits for Offset
- inner length appears a 0 when something is to be inserted or as a positive integer when something is to be deleted or replaced, i.e.
```"failure": "Consecutive blank lines are forbidden",
    "fix": [
      {
        "innerStart": 78,
        "innerLength": 2,
        "innerText": ""
      }
    ],```
or
```"failure": "Forbidden 'var' keyword, use 'let' or 'const' instead",
    "fix": {
      "innerStart": 210,
      "innerLength": 3,
      "innerText": "let"
    },```
- so innerLenght fits DeletedLength
- InnerText and InnerBytes seem to differ only in encoding, which I've fixed
Let me know if you still have concerns about this

Reply:
Sorry for the bad formatting, GitHub seems to have different markdown rules.
For the first example "fix" the failure is "failure": "Consecutive blank lines are forbidden",

Reply:
I think you are right about this.